client/src/App.tsx

```diff
@@ -210,6 +216,90 @@ function App() {
 		}
 	}, []);
 
+	const activateFireworks = () => {
+		var duration = 2 * 1000;
+		var animationEnd = Date.now() + duration;
+		var defaults = { startVelocity: 30, spread: 360, ticks: 60, zIndex: 0 };
+
+		function randomInRange(min: number, max: number) {
+			return Math.random() * (max - min) + min;
+		}
+
+		var interval: any = setInterval(function () {
+			var timeLeft = animationEnd - Date.now();
+
+			if (timeLeft <= 0) {
+				return clearInterval(interval);
+			}
+
+			var particleCount = 50 * (timeLeft / duration);
+			// since particles fall down, start a bit higher than random
+			confetti(
+				Object.assign({}, defaults, {
+					particleCount,
+					origin: { x: randomInRange(0.1, 0.3), y: Math.random() - 0.2 }
+				})
+			);
+			confetti(
+				Object.assign({}, defaults, {
+					particleCount,
+					origin: { x: randomInRange(0.7, 0.9), y: Math.random() - 0.2 }
+				})
+			);
+		}, 250);
+	};
+
+	const activateSnow = async () => {
+		var duration = 3.5 * 1000;
+		var animationEnd = Date.now() + duration;
+		var skew = 1;
+
+		function randomInRange(min: number, max: number) {
+			return Math.random() * (max - min) + min;
+		}
+
+		(function frame() {
```

Comment:
dont think we need an anonymous function here, why not just execute code? adds complexity

Reply:
For this one, I have to write the frame function inside another function because the requestAnimationFrame function from the library does not support callback function with arguments. It only allows callback function without arguments. But if we want to define the frame function outside the activateSchoolPride or ActivateSnow functions, we need to pass the arguments to the frame function for recursion to work properly.  So, defining inside the activae_ function, i don't have to pass arguments.